main/src/ca/uwaterloo/flix/api/lsp/provider/FindReferencesProvider.scala

```diff
@@ -80,10 +80,12 @@ object FindReferencesProvider {
     ("status" -> "success") ~ ("result" -> locs.map(_.toJSON))
   }
 
-  private def findDefUses(sym: Symbol.DefnSym)(implicit index: Index, root: Root): JObject = {
+  private def findDefReferences(sym: Symbol.DefnSym)(implicit index: Index, root: Root): JObject = {
     val uses = index.usesOf(sym)
+    val locationLink = LocationLink.fromDefSym(sym, sym.loc)(root)
+    val defLoc = Location(locationLink.targetUri, locationLink.targetRange)
     val locs = uses.toList.map(Location.from)
-    ("status" -> "success") ~ ("result" -> locs.map(_.toJSON))
+    ("status" -> "success") ~ ("result" -> locs.appended(defLoc).map(_.toJSON))
   }
 
```

Comment:
If you refactor to:

```
    val defn = Location.from(sym)
    val uses = index.usesOf(sym)
    val locs = defn :: uses.toList.map(Location.from)
    ("status" -> "success") ~ ("result" -> locs.map(_.toJSON))
```

then I am happy :)

Reply:
cool :)

Reply:
Following the snippet you posted, I have refactor the code in 2 steps:
* use infix cons operator and reordering
* implement a from(DefnSym) method in Location

I am not sure if you meant the second point, so in case in this way we can simply drop the commit :)

Reply:
You dont need the second part. You can just call `Location.from(sym.loc)` :)